apis/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -51,6 +66,10 @@ type ForeignClusterSpec struct {
 	// +kubebuilder:default="Unknown"
 	// Indicates if this remote cluster is trusted or not.
 	TrustMode discovery.TrustMode `json:"trustMode,omitempty"`
+	// If discoveryType is LAN or WAN and this indicates the number of seconds after that
+	// this ForeignCluster will be removed if no updated has been received.
+	// +kubebuilder:validation:Minimum=0
+	TTL int `json:"ttl,omitempty"`
```

Comment:
Not clear to me whether this field is necessary. Is it possible to be manage this at the parent level (the resource which specifies the DNS domain), to avoid continuously triggering the reconciliation of this resource? As this is an user-facing API I would avoid adding fields that may be misleading (I am also not completely sure whether the DIscoveryType field is required, as well as the ClusterIdentity one, the latter looking to me more part of the status).

Reply:
Ref #670